packages/editor/src/components/blocks/datatable/DataTable.tsx

```diff
@@ -0,0 +1,58 @@
+import type { DataTable, Prettify } from '@axonivy/form-editor-protocol';
+import type { ComponentConfig, UiComponentProps } from '../../../types/config';
+import './DataTable.css';
+import { baseComponentFields, defaultBaseComponent } from '../base';
+import IconSvg from './DataTable.svg?react';
+import { ComponentBlock } from '../../../editor/canvas/ComponentBlock';
+import { DataTableColumnDialog } from '../../../editor/canvas/data-table-column/DataTableColumnDialog';
+import { Message } from '@axonivy/ui-components';
+
+type DataTableProps = Prettify<DataTable>;
+
+export const defaultDataTableProps: DataTable = {
+  components: [],
+  label: 'Label',
+  value: '',
+  ...defaultBaseComponent
+} as const;
+
+export const DataTableComponent: ComponentConfig<DataTableProps> = {
+  name: 'DataTable',
+  category: 'Elements',
+  subcategory: 'Input',
+  icon: <IconSvg />,
+  description: 'A datatable component',
+  defaultProps: defaultDataTableProps,
+  render: props => <UiBlock {...props} />,
+  create: ({ label, value, ...defaultProps }) => ({ ...defaultDataTableProps, label, value, ...defaultProps }),
+  outlineInfo: component => component.label,
+  fields: {
+    components: { subsection: 'General', type: 'hidden' },
+    label: { subsection: 'General', label: 'Label', type: 'text' },
+    value: { subsection: 'General', label: 'Value', type: 'textBrowser', options: { onlyTypesOf: 'List<' } },
+    ...baseComponentFields
+  }
+};
+
+const UiBlock = ({ id, label, components, value }: UiComponentProps<DataTableProps>) => {
+  return (
+    <>
+      <div className='block-table'>
+        <div className='block-table__label'>
+          <span>{label}</span>
+          <span style={{ color: 'var(--N600)' }}>
+            {value.length > 0 ? value : <Message variant='warning' message='Value of DataTable is empty' />}
+          </span>
+        </div>
+        {components.length > 0 && (
+          <div className='block-table__columns'>
+            {components.map((column, index) => {
+              return <ComponentBlock key={column.id} component={column} preId={components[index - 1]?.id} />;
+            })}
+          </div>
+        )}
+        {components.length === 0 && <DataTableColumnDialog id={id} initValue={value} />}
```

Comment:
I wouldn't do this...
Just render a warning which is fine, and if someone wants to define this then go to the property view. -> Keep it simple and stupid

Reply:
Or better just provide a button here if the value is set to create the columns but not a button for the dialog...